python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -1652,28 +1676,6 @@ def fillna(
         Returns
         -------
         DataFrame or Series
-
-        .. pandas-compat::
-            **groupby.fillna**
-
-            This function may return result in different format to the method
-            Pandas supports. For example:
-
-            .. code-block::
-
-                >>> df = pd.DataFrame({'k': [1, 1, 2], 'v': [2, None, 4]})
-                >>> gdf = cudf.from_pandas(df)
-                >>> df.groupby('k').fillna({'v': 4}) # pandas
```

Comment:
AFAICT this is no longer the behavior of pandas and `reset_index` must be called somewhere in pandas internally.

```python
>>> import pandas as pd
>>> pd.__version__
'1.5.3'
>>> df = pd.DataFrame({'k': [1, 1, 2], 'v': [2, None, 4]})
>>> df.groupby('k').fillna({'v': 4})
     v
0  2.0
1  4.0
2  4.0
```